test/ecto/query_test.exs

```diff
@@ -715,4 +715,18 @@ defmodule Ecto.QueryTest do
       end
     end
   end
+
+  describe "reverse_order/1" do
+    test "reverses the order of a simple query" do
+      order_bys = [asc: :inserted_at, desc: :id]
+      reversed_order_bys = [desc: :inserted_at, asc: :id]
+      q = from(p in "posts")
+      assert reverse_order(order_by(q, ^order_bys)) == order_by(q, ^reversed_order_bys)
+    end
+
+    test "is a no-op on a query with no order" do
+      q = from(p in "posts")
+      assert reverse_order(q) == q
```

Comment:
I'm really not sure about this behaviour. It seems very weird to me that `Repo.all(query) == Repo.all(reverse(query))`. I wouldn't expect such a behaviour.